app/plugin.go

```diff
@@ -4,356 +4,133 @@
+func (a *App) SyncPluginsActiveState() {
+	if a.Plugins == nil {
 		return
 	}
 
-	plugins, err := a.PluginEnv.Plugins()
-	if err != nil {
-		mlog.Error(fmt.Sprintf("Cannot setPluginsActive(%t)", activate), mlog.Err(err))
-		return
-	}
-
-	for _, plugin := range plugins {
-		if plugin.Manifest == nil {
-			continue
-		}
-
-		enabled := false
-		if state, ok := a.Config().PluginSettings.PluginStates[plugin.Manifest.Id]; ok {
-			enabled = state.Enable
-		}
-
-		a.pluginStatuses[plugin.Manifest.Id] = &model.PluginStatus{
-			ClusterId:   a.GetClusterId(),
-			PluginId:    plugin.Manifest.Id,
-			PluginPath:  filepath.Dir(plugin.ManifestPath),
-			IsSandboxed: a.IsPluginSandboxSupported,
-			Name:        plugin.Manifest.Name,
-			Description: plugin.Manifest.Description,
-			Version:     plugin.Manifest.Version,
-		}
-
-		if activate && enabled {
-			a.setPluginActive(plugin, activate)
-		} else if !activate {
-			a.setPluginActive(plugin, activate)
-		}
-	}
-
-	if err := a.notifyPluginStatusesChanged(); err != nil {
-		mlog.Error("failed to notify plugin status changed", mlog.Err(err))
-	}
-}
-
-func (a *App) setPluginActiveById(id string, activate bool) {
-	plugins, err := a.PluginEnv.Plugins()
-	if err != nil {
-		mlog.Error(fmt.Sprintf("Cannot setPluginActiveById(%t)", activate), mlog.String("plugin_id", id), mlog.Err(err))
-		return
-	}
+	config := a.Config().PluginSettings
 
-	for _, plugin := range plugins {
-		if plugin.Manifest != nil && plugin.Manifest.Id == id {
-			a.setPluginActive(plugin, activate)
+	if *config.Enable {
+		availablePlugins, err := a.Plugins.Available()
+		if err != nil {
+			a.Log.Error("Unable to get available plugins", mlog.Err(err))
+			return
 		}
-	}
-}
-
-func (a *App) setPluginActive(plugin *model.BundleInfo, activate bool) {
-	if plugin.Manifest == nil {
-		return
-	}
-
-	id := plugin.Manifest.Id
-
-	active := a.PluginEnv.IsPluginActive(id)
 
-	if activate {
-		if !active {
-			if err := a.activatePlugin(plugin.Manifest); err != nil {
-				mlog.Error("Plugin failed to activate", mlog.String("plugin_id", plugin.Manifest.Id), mlog.String("err", err.DetailedError))
+		// Deactivate any plugins that have been disabled.
+		for _, plugin := range a.Plugins.Active() {
+			// Determine if plugin is enabed
+			pluginId := plugin.Manifest.Id
+			pluginEnabled := false
+			if state, ok := config.PluginStates[pluginId]; ok {
+				pluginEnabled = state.Enable
 			}
-		}
 
-	} else if !activate {
-		if active {
-			if err := a.deactivatePlugin(plugin.Manifest); err != nil {
-				mlog.Error("Plugin failed to deactivate", mlog.String("plugin_id", plugin.Manifest.Id), mlog.String("err", err.DetailedError))
-			}
-		} else {
-			if err := a.setPluginStatusState(plugin.Manifest.Id, model.PluginStateNotRunning); err != nil {
-				mlog.Error("Plugin status state failed to update", mlog.String("plugin_id", plugin.Manifest.Id), mlog.String("err", err.Error()))
+			// If it's not enabled we need to stop it
```

Comment:
nit: "stop" vs. "deactivate". Maybe `// Deactivate plugin if not enabled`? Or just remove the comments altogether, since this code is pretty self-explanatory.